src/diracx/routers/job_manager/sandboxes.py

```diff
@@ -192,3 +193,42 @@ async def get_sandbox_file(
     return SandboxDownloadResponse(
         url=presigned_url, expires_in=settings.url_validity_seconds
     )
+
+
+@router.get("/jobs/{job_id}/sandbox")
+async def get_job_sandboxes(
+    job_id: int,
+    sandbox_metadata_db: SandboxMetadataDB,
+) -> dict[str, list[Any]]:
+    """Get input and output sandboxes of given job id at the same time."""
+    # TODO: check that user as created the job or is admin
+    input_sb = await sandbox_metadata_db.get_sandbox_assigned_to_job(job_id, "Input")
+    output_sb = await sandbox_metadata_db.get_sandbox_assigned_to_job(job_id, "Output")
+    return {"Input": input_sb, "Output": output_sb}
+
+
+@router.get("/jobs/{job_id}/sandbox/{sandbox_type}")
+async def get_job_sandbox(
+    job_id: int,
+    sandbox_metadata_db: SandboxMetadataDB,
+    sandbox_type: Literal["input", "output"],
+) -> list[Any]:
+    """Get input or output sandbox from given job"""
+    # TODO: check that user has created the job or is admin
+    job_sb_pfns = await sandbox_metadata_db.get_sandbox_assigned_to_job(
+        job_id, sandbox_type.capitalize()
+    )
+    return job_sb_pfns
+
+
+@router.patch("/jobs/{job_id}/sandbox/output")
+async def assign_sandbox(
+    job_id: int,
+    pfn: Annotated[str, Query(max_length=256, pattern=SANDBOX_PFN_REGEX)],
```

Comment:
I think the 404 problem comes from here. the `pfn` is not in the `query` but in the `body`

Reply:
Thanks @chaen , I simply try that: https://github.com/DIRACGrid/diracx/pull/156/commits/c968b65ac330cf6ce9b7fc66c2b663a2f72907c7
I've tried the Pydantic syntax as well, but it doesn't solve the issue... I also checked that the job and the sandbox are created in the test and it seems to be the case.

Reply:
Yes, I modified the `sandbox_pfn` declaration. I've investigated but I didn't find any obvious solution. Could be helpful if you can have a look :)

Reply:
OK so the problem was a bit more subtle :-)
You declare your routes like this for example

```python
@router.get("/jobs/{job_id}/sandbox")
```
However, the `/jobs` prefix is already automatically prepended when creating the `job` router (this comes from that, and yes, that is part of the missing docs :-) ) https://github.com/DIRACGrid/diracx/blob/3b4761c5c96c4c52c706196450c01e8765855b32/setup.cfg#L88

So the way you need to declare your routes is not `/jobs/{job_id}/sandbox" but just "/{job_id}/sandbox"

Effectively, your route is available as "/jobs/jobs/{job_id}/sandbox"

So changing the route definition fixes the problem you have, but you bump then into another one, which I let in your care :-)

Reply:
Thanks a lot @chaen! 
Indeed I could see that when looking other sandbox routes... I also fixed the `pfn` parameter by splitting it inside the `assign_sandbox` route. Is that the right way to do it ? 
Otherwise the tests finally passed, the PR should be valid.